lib/watch-sidebar.js

```diff
@@ -102,17 +102,41 @@ export default class WatchSidebar {
   }
 
   removeWatch() {
-    const watches = this.watchViews.map((v, k) => ({
-      name: v.getCode(),
-      value: k
-    }));
-    WatchesPicker.onConfirmed = item => {
-      this.watchViews[item.value].destroy();
-      this.watchViews.splice(item.value, 1);
-      if (this.watchViews.length === 0) this.addWatch();
-    };
-    WatchesPicker.setItems(watches);
-    WatchesPicker.toggle();
+    const watches = this.watchViews
+      .map((v, k) => ({
+        name: v.getCode(),
+        value: k
+      }))
+      .filter(obj => obj.value !== 0 || obj.name !== "");
+
```

Comment:
We always keep one empty string watch if no other watches, but the picker should show empty if this is the only watch left. You should be able to remove empty watches other than the placeholder mentioned, see gif for examples of what I am saying:

![last-element](https://cloud.githubusercontent.com/assets/10860657/25309276/81c50b82-278e-11e7-89b8-fac5d8bd3eb9.gif)